src/Main.java

```diff
@@ -0,0 +1,228 @@
+import java.util.ArrayList;
+
+interface IBoardGame {
+	public char getWinner();
+	public boolean takeSlot(boolean isX, int row, int column);
+}
+
+class XOBoardGame implements IBoardGame {
+	private ArrayList<ArrayList<Character>> board;
+	private int boardSize;
+	
+	// a count of the capture on each row
+	private ArrayList<Integer> xVertical;
+	private ArrayList<Integer> oVertical;
+	
+	// a count of the capture on each column
+	private ArrayList<Integer> xHorizontal;
+	private ArrayList<Integer> oHorizontal;
+	
+	// a count of the capture on negative diagonal line (/)
+	// the capture will be on negative diagonal line when row + column = n-1
+	// example (n = 3): (0, 2) (1, 1) (2, 0)
+	private int xNegativeDiagonal;
+	private int oNegativeDiagonal;
+	
+	// a count of the capture on positive diagonal line (\)
+	// the capture will be on positive diagonal line when row - column = 0
+	// example (n = 3): (0, 0) (1, 1) (2, 2)
+	private int xPositiveDiagonal;
+	private int oPositiveDiagonal;
+	
+	private char theWinner;
+	
+	@Override
+	public char getWinner() {
+		return theWinner;
+	}
+	
+	public XOBoardGame(int boardSize) {
+		this.boardSize = boardSize;
+		
+		initializeBoard();
+		initializeCaptureCount();
+		printBoard();
+	}
+	
+	private void initializeBoard() {
+		board = new ArrayList<ArrayList<Character>>();
+		
+		for (int i = 0 ; i < boardSize ; ++i) {
+			board.add(new ArrayList<Character>());
+			
+			for (int j = 0 ; j < boardSize ; ++j) {
+				board.get(i).add(' ');
+			}
+		}
+	}
+	
+	private void initializeCaptureCount() {
+		xVertical = new ArrayList<Integer>();
+		oVertical = new ArrayList<Integer>();
+		
+		xHorizontal = new ArrayList<Integer>();
+		oHorizontal = new ArrayList<Integer>();
+		
+		for (int i = 0 ; i < boardSize ; ++i) {
+			xVertical.add(0);
+			oVertical.add(0);
+			
+			xHorizontal.add(0);
+			oHorizontal.add(0);
+		}
+	}
+	
+	private void printBoard() {
```

Comment:
ควรจะแยกเรื่องของการแสดงผลออกจากการคำนวณของตัวเกมครับ เพราะถ้าเกมนี้เปลี่ยนไปเล่นบนเว็บ หรือบนมือถือ แอพเราจะต้องถูกรื้อแก้ใหม่ เป็นหลักของ Separation of concerns ครับ

Reply:
ก็คือประมาณว่าทำให้ XOBoardGame เป็น Abstract Class และ printBoard เป็น Abstract Method ใช่ไหมครับ แล้วก็ extends ไปขยายความใน Child Class เช่น XOBoardGamePC, XOBoardGameMobile ประมาณนี้ใช่ไหมอ่าครับ